odoo/models.py

```diff
@@ -1755,7 +1764,7 @@ def name_search(self, name='', args=None, operator='ilike', limit=100):
         return [(record.id, record.display_name) for record in records.sudo()]
 
     @api.model
-    def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None):
+    def _name_search(self, name, domain=None, operator='ilike', limit=None, order=None) -> Query:
```

Comment:
This is not true for overrides and we will remove this function soon.
You can skip name search for now.